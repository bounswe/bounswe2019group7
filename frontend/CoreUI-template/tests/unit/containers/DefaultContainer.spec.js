import BootstrapVue from 'bootstrap-vue'
import { shallowMount, createLocalVue } from '@vue/test-utils';
import VueRouter from 'vue-router'
import DefaultContainer from '@/containers/DefaultContainer'

const localVue = createLocalVue()
localVue.use(VueRouter)
const router = new VueRouter()

localVue.use(BootstrapVue)

describe('DefaultContainer.vue', () => {
  it('has a name', () => {
    expect(DefaultContainer.name).toMatch('DefaultContainer')
  })
  it('has a created hook', () => {
    expect(typeof DefaultContainer.data).toMatch('function')
  })
  test('renders correctly', () => {
    const wrapper = shallowMount(DefaultContainer, {
      localVue,
        router
    })
    expect(wrapper.element).toMatchSnapshot()
  })
  it('sets the correct default data', () => {
    expect(typeof DefaultContainer.data).toMatch('function')
    const defaultData = DefaultContainer.data()
    expect(typeof defaultData.nav).toMatch('object')
  })
  it('is Vue instance', () => {
    const wrapper = shallowMount(DefaultContainer, {
      localVue,
      router
    })
    expect(wrapper.isVueInstance()).toBe(true)
  })
})
