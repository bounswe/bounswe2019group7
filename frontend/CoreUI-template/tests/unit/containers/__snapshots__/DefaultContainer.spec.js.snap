// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultContainer.vue renders correctly 1`] = `
<div
  class="app"
>
  <appheader-stub
    fixed="true"
  >
    <sidebartoggler-stub
      class="d-lg-none"
      defaultopen="true"
      display="md"
      mobile="true"
    />
     
    <b-link-stub
      class="navbar-brand"
      event="click"
      routertag="a"
      target="_self"
      to="#"
    >
      <img
        alt="CoreUI Logo"
        class="navbar-brand-full"
        height="25"
        src="img/brand/logo.svg"
        width="89"
      />
       
      <img
        alt="CoreUI Logo"
        class="navbar-brand-minimized"
        height="30"
        src="img/brand/sygnet.svg"
        width="30"
      />
    </b-link-stub>
     
    <sidebartoggler-stub
      class="d-md-down-none"
      defaultopen="true"
      display="lg"
    />
     
    <b-navbar-nav-stub
      class="d-md-down-none"
      tag="ul"
    >
      <b-nav-item-stub
        class="px-3"
        event="click"
        linkattrs="[object Object]"
        routertag="a"
        target="_self"
        to="/dashboard"
      >
        Dashboard
      </b-nav-item-stub>
       
      <b-nav-item-stub
        class="px-3"
        event="click"
        exact="true"
        linkattrs="[object Object]"
        routertag="a"
        target="_self"
        to="/users"
      >
        Users
      </b-nav-item-stub>
       
      <b-nav-item-stub
        class="px-3"
        event="click"
        linkattrs="[object Object]"
        routertag="a"
        target="_self"
      >
        Settings
      </b-nav-item-stub>
    </b-navbar-nav-stub>
     
    <b-navbar-nav-stub
      class="ml-auto"
      tag="ul"
    >
      <b-nav-item-stub
        class="d-md-down-none"
        event="click"
        linkattrs="[object Object]"
        routertag="a"
        target="_self"
      >
        <i
          class="icon-bell"
        />
         
        <b-badge-stub
          event="click"
          pill="true"
          routertag="a"
          tag="span"
          target="_self"
          variant="danger"
        >
          5
        </b-badge-stub>
      </b-nav-item-stub>
       
      <b-nav-item-stub
        class="d-md-down-none"
        event="click"
        linkattrs="[object Object]"
        routertag="a"
        target="_self"
      >
        <i
          class="icon-list"
        />
      </b-nav-item-stub>
       
      <b-nav-item-stub
        class="d-md-down-none"
        event="click"
        linkattrs="[object Object]"
        routertag="a"
        target="_self"
      >
        <i
          class="icon-location-pin"
        />
      </b-nav-item-stub>
       
      <defaultheaderdropdownaccnt-stub />
    </b-navbar-nav-stub>
     
    <asidetoggler-stub
      class="d-none d-lg-block"
      display="lg"
    />
  </appheader-stub>
   
  <div
    class="app-body"
  >
    <appsidebar-stub
      fixed="true"
    >
      <sidebarheader-stub />
       
      <sidebarform-stub />
       
      <sidebarnav-stub
        navitems="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"
      />
       
      <sidebarfooter-stub />
       
      <sidebarminimizer-stub />
    </appsidebar-stub>
     
    <main
      class="main"
    >
      <breadcrumb-stub
        list=""
      />
       
      <div
        class="container-fluid"
      >
        <router-view-stub
          name="default"
        />
      </div>
    </main>
     
    <appaside-stub
      display=""
      fixed="true"
      offcanvas="true"
    >
      <defaultaside-stub />
    </appaside-stub>
  </div>
   
  <thefooter-stub>
    <div>
      <a
        href="https://coreui.io"
      >
        CoreUI
      </a>
       
      <span
        class="ml-1"
      >
        Â© 2018 creativeLabs.
      </span>
    </div>
     
    <div
      class="ml-auto"
    >
      <span
        class="mr-1"
      >
        Powered by
      </span>
       
      <a
        href="https://coreui.io"
      >
        CoreUI for Vue
      </a>
    </div>
  </thefooter-stub>
</div>
`;
